kalite/topic_tools/__init__.py

```diff
@@ -45,12 +44,15 @@
 # Globals that can be filled
 TOPICS          = None
 CACHE_VARS.append("TOPICS")
-def get_topic_tree(force=False, annotate=False, channel=settings.CHANNEL):
+def get_topic_tree(force=False, annotate=False, channel=settings.CHANNEL, language="en"):
+    translation.activate(language)
     global TOPICS, TOPICS_FILEPATHS
     if not TOPICS:
         TOPICS = {}
     if TOPICS.get(channel) is None:
-        TOPICS[channel] = softload_json(TOPICS_FILEPATHS.get(channel), logger=logging.debug, raises=False)
+        TOPICS[channel] = {}
+    if TOPICS.get(channel, {}).get(language) is None:
```

Comment:
I'm trying to think of negative consequences to this approach, the only one I can think of right now is that there might be a bit more memory bloat in storing an entire topic as part of the cache tree vs. just regenerating the topic tree on the fly (if i remember how the old one was done correctly).